### PR TITLE
docs/development_guide: update Helm chart guide

### PR DESCRIPTION
**Pull request by shashankram**
Updates the Helm section in the development guide.

### CODE REVIEW

File: docs/development_guide.md [hunk 1]
```diff
@@ -167,11 +167,23 @@ TODO:
...
-subchart which can be found in the [`charts/partials`](charts/partials) folder.
+The Open Service Mesh control plane charts are located in the
+[`charts/osm`](../charts/osm) folder.
+
+The [`charts/osm/values.yaml`](../charts/osm/values.yaml) file defines the default value for properties
+referenced by the different chart templates.
+
+The [`charts/osm/templates/`](../charts/osm/templates/) folder contains the chart templates
+for the different Kubernetes resources that are deployed as a part of the Open Service control plane installation.
+The different chart templates are used as follows.
```
Comment (snehachhabria) [inline docs/development_guide.md:178]:
```suggestion
The different chart templates used are as follows:
```

File: docs/development_guide.md [hunk 2]
```diff
@@ -167,11 +167,23 @@ TODO:
...
+- `osm-*.yaml` chart templates are directly consumed by the `osm-controller` service.
+- `mutatingwebhook.yaml` is used to deploy the `MutatingWebhookConfiguration` kubernetes resource that enables automatic sidecar injection
```
Comment (snehachhabria) [inline docs/development_guide.md:180]:
```suggestion
- `mutatingwebhook.yaml` is used to deploy a `MutatingWebhookConfiguration` kubernetes resource that enables automatic sidecar injection
```

File: docs/development_guide.md [hunk 3]
```diff
@@ -167,11 +167,23 @@ TODO:
...
+-  `grafana-*.yaml` chart templates are used to deploy the Grafana instance when the metrics stack is enabled
```
Comment (snehachhabria) [inline docs/development_guide.md:181]:
```suggestion
-  `grafana-*.yaml` chart templates are used to deploy a Grafana instance when the metrics stack is enabled
```

File: docs/development_guide.md [hunk 4]
```diff
@@ -167,11 +167,23 @@ TODO:
...
+- `prometheus-*.yaml` chart templates are used to deploy the Prometheus instance when the metrics stack is enabled
```
Comment (snehachhabria) [inline docs/development_guide.md:182]:
```suggestion
- `prometheus-*.yaml` chart templates are used to deploy a Prometheus instance when the metrics stack is enabled
```

File: docs/development_guide.md [hunk 5]
```diff
@@ -167,11 +167,23 @@ TODO:
...
+- `zipkin-*.yaml` chart templates are used to deploy the Zipkin instance when Zipkin tracing is enabled
```
Comment (snehachhabria) [inline docs/development_guide.md:183]:
```suggestion
- `zipkin-*.yaml` chart templates are used to deploy a Zipkin instance when Zipkin tracing is enabled
```

File: docs/development_guide.md [hunk 6]
```diff
@@ -167,11 +167,23 @@ TODO:
...
 
-The Open Service Mesh control plane chart is located in the
-[`charts/openservicemesh`](charts/openservicemesh) folder. The [`charts/patch`](charts/patch)
-chart consists of the Open Service Mesh proxy specification, which is used by the proxy
-injector to inject the proxy container. Both charts depend on the partials
-subchart which can be found in the [`charts/partials`](charts/partials) folder.
+The Open Service Mesh control plane charts are located in the
+[`charts/osm`](../charts/osm) folder.
+
+The [`charts/osm/values.yaml`](../charts/osm/values.yaml) file defines the default value for properties
```
Comment (nojnhuh) [inline docs/development_guide.md:173]: Would this description about the contents of the chart fit better in the README at the root of the chart?

File: docs/development_guide.md [hunk 7]
```diff
@@ -167,9 +167,24 @@ TODO:
 
 ## Helm chart
 
-The Open Service Mesh control plane chart is located in the
+The Open Service Mesh control plane charts are located in the
 [`charts/osm`](../charts/osm) folder.
```
Comment (nojnhuh) [inline docs/development_guide.md:171]: Links to files can be absolute based on the root of the repo, so then this link won't break if we move this document:
```suggestion
[`charts/osm`](/charts/osm) folder.
```

File: docs/development_guide.md [hunk 8]
```diff
@@ -167,9 +167,24 @@ TODO:
 
 ## Helm chart
 
-The Open Service Mesh control plane chart is located in the
+The Open Service Mesh control plane charts are located in the
```
Comment (nojnhuh) [inline docs/development_guide.md:170]: The whole directory is called a "chart," so the way it was before makes more sense.

File: docs/development_guide.md (hunk 6 above)
Comment (shashankram) [reply docs/development_guide.md:173]: We could do it that way. I was thinking it makes sense to be in the helm section of the development guide. WDYT?

Comment (nojnhuh) [reply docs/development_guide.md:173]: Yeah, I think it does make more sense here. If we publish the chart standalone, we wouldn't need those kinds of descriptions.

File: docs/development_guide.md (hunk 8 above)
Comment (shashankram) [reply docs/development_guide.md:170]: Updating.

File: docs/development_guide.md [hunk 9]
```diff
@@ -168,7 +168,22 @@ TODO:
...
+[`charts/osm`](/charts/osm) folder.
+
+The [`charts/osm/values.yaml`](/charts/osm/values.yaml) file defines the default value for properties
+referenced by the different chart templates.
+
+The [`charts/osm/templates/`](/charts/osm/templates/) folder contains the chart templates
+for the different Kubernetes resources that are deployed as a part of the Open Service control plane installation.
+The different chart templates are used as follows:
+- `osm-*.yaml` chart templates are directly consumed by the `osm-controller` service.
+- `mutatingwebhook.yaml` is used to deploy the `MutatingWebhookConfiguration` kubernetes resource that enables automatic sidecar injection
```
Comment (snehachhabria) [inline docs/development_guide.md:180]:
```suggestion
- `mutatingwebhook.yaml` is used to deploy a `MutatingWebhookConfiguration` kubernetes resource that enables automatic sidecar injection
```

**Pull request merged**